hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HiveSchemaUtil.java

```diff
@@ -429,12 +429,12 @@ public static boolean isSchemaTypeUpdateAllowed(String prevType, String newType)
     newType = newType.toLowerCase();
     if (prevType.equals(newType)) {
       return true;
-    } else if (prevType.equalsIgnoreCase(INT_TYPE_NAME) && newType.equalsIgnoreCase(BIGINT_TYPE_NAME)) {
-      return true;
-    } else if (prevType.equalsIgnoreCase(FLOAT_TYPE_NAME) && newType.equalsIgnoreCase(DOUBLE_TYPE_NAME)) {
-      return true;
+    } else if (prevType.equalsIgnoreCase(STRING_TYPE_NAME) && newType.equalsIgnoreCase(BINARY_TYPE_NAME)) {
+      return false;
+    } else if (prevType.equalsIgnoreCase(BINARY_TYPE_NAME) && newType.equalsIgnoreCase(STRING_TYPE_NAME)) {
+      return false;
     } else {
-      return prevType.contains("struct") && newType.toLowerCase().contains("struct");
+      return true;
     }
```

Comment:
So a bigint can promote to int?

Reply:
nope, but since hudi itself does not support bigint -> int, then there was no reason to check this case